pkg/compute/executor.go

```diff
@@ -74,10 +74,13 @@ func PrepareRunArguments(
 		return nil
 	})
 
-	var engineArgs interface{}
-	engineArgs = execution.Job.Spec.Docker
-	if execution.Job.Spec.Engine == model.EngineWasm {
-		importModuleVolumes, err := storage.ParallelPrepareStorage(ctx, strgprovider, execution.Job.Spec.Wasm.ImportModules...)
+	var engineArgs *executor.Arguments
+	if execution.Job.Spec.EngineSpec.Type == model.EngineWasm.String() {
```

Comment:
What's the more general solution to this problem?

Reply:
Yes, good idea, will add a comment.

The more general solution is make the WASM executor aware of which fields in the `spec.Inputs` StorageSpec are WASM modules. We would need to alter the WASM `EngineSpec` such that it can reference values from `spec.Inputs` with its `EntryModule` and `ImportModules` fields. 
(I suspect future implementations of an `EngineSpec` will need this ability - referencing specific inputs via their arguments - docker image comes to mind as a potential candidate).

In https://github.com/bacalhau-project/bacalhau/pull/2675 we modified the Compute Node to initialize and download all `spec.Input`s to local storage before passing it to the executor. Previously executors we responsible for downloading their inputs to local storage, and running the job. With our shift towards pluggable executors in https://github.com/bacalhau-project/bacalhau/pull/2637 configuring executor plugins to handle the download of different storage specs seems impractical (@wdbaruni's [comment](https://github.com/bacalhau-project/bacalhau/pull/2637#issuecomment-1625739030) provides more context on the need for the change).